lib/errors/error-collector.js

```diff
@@ -89,69 +89,47 @@ class ErrorCollector {
   }
 
   /**
-   * Helper method for processing errors that are created with .noticeError()
+   * Gets the iterable property from the transaction based on the error type
+   * @param {Transaction} transaction the collected exception's transaction
+   * @param {string} errorType the type of error to be processed; "user", "transactionException", "transaction"
+   * @returns
+   */
+  _getIterableProperty(transaction, errorType) {
+    let iterableProperty = null
+    if (errorType === 'user' && transaction.userErrors.length) {
+      iterableProperty = transaction.userErrors
+    }
+    if (errorType === 'transactionException') {
+      iterableProperty = transaction.exceptions
+    }
+    return iterableProperty
+  }
+
+  /**
+   * Helper method for processing errors that are created with .noticeError(), exceptions
+   * on the transaction (transaction.exceptions array), and inferred errors based on Transaction metadata.
    *
    * @param {Transaction} transaction the collected exception's transaction
-   * @param {number} collectedErrors the number of errors we've successfully .collect()-ed
+   * @param {number} collectedErrors the number of errors we've succesfully .collect()-ed
    * @param {number} expectedErrors the number of errors marked as expected in noticeError
-   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post-processing
+   * @param {string} errorType the type of error to be processed; "user", "transactionException", "transaction"
```

Comment:
```suggestion
   * @param {string} errorType the type of error to be processed: "user", "transactionException", "transaction"
```

Reply:
I'm suggesting a colon here, since we're introducing a list of potential error types in the JSDoc line here, rather than joining two independent clauses.